addons/MountMuzzle/mountmuzzle.lua

```diff
@@ -0,0 +1,222 @@
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name     = 'Mount Muzzle'
+_addon.author   = 'Sjshovan (Apogee) sjshovan@gmail.com'
+_addon.version  = '0.9.2'
+_addon.commands = {'mountmuzzle', 'muzzle', 'mm'}
+
+local _logger = require('logger')
+local _config  = require('config')
+local _packets = require('packets')
+
+require('helpers')
+
+local mounted = false
+local needs_inject = false
+
+local defaults = {
+    muzzle = muzzles.silent.name
+}
+
+local settings = _config.load(defaults)
+
+local help = {
+    commands = {
+        buildHelpSeperator('=', 26),
+        buildHelpTitle('Commands'),
+        buildHelpSeperator('=', 26),
+        buildHelpCommandEntry('list', 'Display the available muzzle types.'),
+        buildHelpCommandEntry('set <muzzle>', 'Set the current muzzle to the given muzzle type.'),
+        buildHelpCommandEntry('get', 'Display the current muzzle.'),
+        buildHelpCommandEntry('default', 'Set the current muzzle to the default (Silent).'),
+        buildHelpCommandEntry('reload', 'Reload Mount Muzzle.'),
+        buildHelpCommandEntry('help', 'Display Mount Muzzle commands.'),
+        buildHelpSeperator('=', 26),
+    },
+
+    types = {
+        buildHelpSeperator('=', 23),
+        buildHelpTitle('Types'),
+        buildHelpSeperator('=', 23),
+        buildHelpTypeEntry(ucFirst(muzzles.silent.name), muzzles.silent.description),
+        buildHelpTypeEntry(ucFirst(muzzles.normal.name), muzzles.normal.description),
+        buildHelpTypeEntry(ucFirst(muzzles.choco.name), muzzles.choco.description),
+        buildHelpTypeEntry(ucFirst(muzzles.zone.name), muzzles.zone.description),
+        buildHelpSeperator('=', 23),
+    },
+}
+
+function display_help(table_help)
+    for index, command in pairs(table_help) do
+        displayResponse(command)
+    end
+end
+
+function getMuzzle()
+    return settings.muzzle
+end
+
+function setMuzzle(muzzle)
+    settings.muzzle = muzzle
+    settings:save()
+end
+
+function muzzleValid(muzzle)
+    return muzzles[muzzle] ~= nil
+end
+
+function resolveCurrentMuzzle()
+    local current_muzzle = getMuzzle()
+    if not muzzleValid(current_muzzle) then
+        current_muzzle = muzzles.silent.name
+        setMuzzle(current_muzzle)
+        displayResponse(
+            'Note: Muzzle found in settings was not valid and is now set to the default (%s).':format('Silent':color(colors.secondary)),
+            colors.warn
+        )
+    end
+    return muzzles[current_muzzle]
+end
+
+function requestInject()
+    needs_inject = true
+end
+
+function injectMuzzleMusic()
+    windower.packets.inject_incoming(
```

Comment:
You can use the packets library to do the packing and conversions for you:
```lua
    packets.inject(packets.new('incoming', packets.inbound.music_change.id, {
        ['BGM Type'] = music.types.mount,
        ['Song ID'] = resolveCurrentMuzzle().song,
    }))
```

Reply:
Yes! Have been using this for next mod, thank you! Done.